src/backend/distributed/sql/udfs/fix_partition_shard_index_names/10.2-3.sql

```diff
@@ -0,0 +1,28 @@
+CREATE FUNCTION pg_catalog.fix_partition_shard_index_names(table_name regclass)
+  RETURNS void
+  LANGUAGE C STRICT
+  AS 'MODULE_PATHNAME', $$fix_partition_shard_index_names$$;
+COMMENT ON FUNCTION pg_catalog.fix_partition_shard_index_names(table_name regclass)
+  IS 'fix index names on partition shards of given table';
+
+CREATE OR REPLACE FUNCTION pg_catalog.fix_all_partition_shard_index_names()
+  RETURNS SETOF regclass
+  LANGUAGE plpgsql
+  AS $$
+DECLARE
+	dist_partitioned_table_oid regclass;
+BEGIN
+  FOR dist_partitioned_table_oid IN SELECT c.oid
+                                    FROM pg_dist_partition p
+                                    JOIN pg_class c ON p.logicalrelid = c.oid
+                                    WHERE c.relkind = 'p'
+		                                ORDER BY c.relname, c.oid
+    LOOP
+      EXECUTE 'SELECT fix_partition_shard_index_names( ' || quote_literal(dist_partitioned_table_oid) || ' )';
```

Comment:
Minor" When I first read this function,  I was surprised to see we quote_literal the `dist_partitioned_table_oid` which is just `c.oid`.

Can we make this a little more explicit:

```
for dist_partitioned_table_name 
    IN SELECT p.logicalrelid ...
```